README.md

```diff
@@ -0,0 +1,117 @@
+
+For this test works perfectly we need to use Kodein to some magic happens. If you continue reading the class at the end of the file you can find the Kodein provisioning dependencies.
+
+```java
+    val testDependencies = Module(allowSilentOverride = true) {
+        bind<ApplicationScopeClass>() with instance(applicationScopeClass)
+        bind<ActivityScopeClass>() with instance(activityScopeClass)
+    }
+```
+
+we create a Module that allows overriding the graph, and we create provisioning for those classes. Depends on your objective you can use instance or provision method. In our case, we used the instance because we are initing Mockito in the test environment setup.
+
+The final part overrides the production graph for the testing ones if you read the setup method you can find how we are doing that.
+
+```java
+   @Before
+    fun setup() {
+        MockitoAnnotations.initMocks(this)
+        val app = InstrumentationRegistry.getInstrumentation().targetContext.asApp()
+        app.resetInjection()
+        app.overrideModule = testDependencies
+    }
+```
+
+First of all, we are initiating all the provisioning mocks with Mockito. After that, we use an extension method that we have in the sample Android application to obtain the application. First, we clean the graph, we like have all dependencies clean before each test, to be repeatable. At last, we save our test module to replace the regular module dependencies.
+
+We need to save the module because we don't have control over the injections. the order of modules has been injected is important. We like that first injected the application modules, after that the activity modules and in the end, the testing modules overriding the previous provisions. When we invoke the start activity, Kodein is going to create the module and injected it in the graph, and we need be ready to apply our testing replaces.
+
+If you check ```KodeinSampleApp.kt```
+
+```java
+class KodeinSampleApp : Application(), KodeinAware {
+    override val kodein = ConfigurableKodein(mutable = true)
+    var overrideModule: Module? = null
+
+    override fun onCreate() {
+        super.onCreate()
+        resetInjection()
+    }
+
+    fun addModule(activityModules: Module) {
+        kodein.addImport(activityModules, true)
+        if (overrideModule != null) {
+            kodein.addImport(overrideModule!!, true)
+        }
+    }
+
+    fun resetInjection() {
+        kodein.clear()
+        kodein.addImport(appDependencies(), true)
+    }
+
+    private fun appDependencies(): Module {
+        return Module(allowSilentOverride = true) {
+            bind<ApplicationScopeClass>() with singleton {
+                ApplicationScopeClass()
+            }
+        }
+    }
+}
+```
+
+We have an instance of Kodein that allows us to have to mutate the graph and compose in different modules.
```

Comment:
Based on the previous code we have got an instance of Kodein configured and ready to override modules in a composable way.

Reply:
done.